src/ophyd_async/epics/adsimdetector/_sim.py

```diff
@@ -21,8 +21,8 @@ def __init__(
         config_sigs: Sequence[SignalR] = (),
         plugins: dict[str, adcore.NDPluginBaseIO] | None = None,
     ):
-        driver = SimDriverIO(prefix + drv_suffix)
-        controller = SimController(driver)
+        driver: SimDriverIO = SimDriverIO(prefix + drv_suffix)
+        controller: SimController = SimController(driver)
```

Comment:
I'm surprised these were needed, can't pyright infer their type here without them?

Reply:
Yes, it infers them fine. I forget why I did this.